### ISSUE TITLE
ILLEGAL_STATE Unknown RESP type 5 followed by java.nio.channels.ClosedChannelException exceptions

### ISSUE THREAD

**Issue by sg002-reactive-prog** [vert-x3/vertx-redis-client#233]
### Version

3.8.3

### Context

We encountered this strange behaviour with the client in production
Basically we us rxMgetMany(keys) method on the redisClient

Max number of calls done to Mget is 60 per second with max keys 33, which is reasonable load

This Mget call suddenly started taking 17 seconds with this exception
io.vertx.redis.client.impl.types.ErrorType: ILLEGAL_STATE Unknown RESP type 5

All subsequent calls to redis started failing with this error
ava.nio.channels.ClosedChannelException: null
at io.netty.handler.stream.ChunkedWriteHandler.discard(ChunkedWriteHandler.java:192) ~[subs.jar:na]
at io.netty.handler.stream.ChunkedWriteHandler.doFlush(ChunkedWriteHandler.java:202) ~[subs.jar:na]
at io.netty.handler.stream.ChunkedWriteHandler.flush(ChunkedWriteHandler.java:133) ~[subs.jar:na]
at io.netty.channel.AbstractChannelHandlerContext.invokeFlush0(AbstractChannelHandlerContext.java:749) ~[subs.jar:na]
at io.netty.channel.AbstractChannelHandlerContext.invokeWriteAndFlush(AbstractChannelHandlerContext.java:764) ~[subs.jar:na]
at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:789) ~[subs.jar:na]
at io.netty.channel.AbstractChannelHandlerContext.writeAndFlush(AbstractChannelHandlerContext.java:757) ~[subs.jar:na]
at io.vertx.core.net.impl.ConnectionBase.write(ConnectionBase.java:119) ~[subs.jar:na]
at io.vertx.core.net.impl.ConnectionBase.writeToChannel(ConnectionBase.java:161) ~[subs.jar:na]
at io.vertx.core.net.impl.NetSocketImpl.writeMessage(NetSocketImpl.java:133) ~[subs.jar:na]
at io.vertx.core.net.impl.NetSocketImpl.write(NetSocketImpl.java:177) ~[subs.jar:na]
at io.vertx.core.net.impl.NetSocketImpl.write(NetSocketImpl.java:171) ~[subs.jar:na]
at io.vertx.redis.client.impl.RedisClient.lambda$send$10(RedisClient.java:259) ~[subs.jar:na]
at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369) ~[subs.jar:na]
at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38) ~[subs.jar:na]
at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163) ~[subs.jar:na]
at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:510) ~[subs.jar:na]
at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:518) ~[subs.jar:na]

The application was unable to connect to redis until it was restarted. All calls to redis were failing.

### Extra
The application is deployed in kubernetes. There were no issues with the redis cluster as another application is also connecting to the same cluster and no issues seen there

Any pointers around this would be greatly appreciated, thanks.

**Comment by pmlopes** [comment]
This is a problem detected at the parser, it probably isn't related to load. You mention 33 entry mget calls. Can you try to change the options config `maxNestedArrays` to a bigger number, say 64 or 128? By default it's 32 and it could be forcing the parser to crash if not enough slots are available.

**Comment by sg002-reactive-prog** [comment]
Thanks for the response. Much appreciated. I will try setting that option.

**Comment by sg002-reactive-prog** [comment]
There are two RedisOptions classes. One in io.vertx.redis package and other in io.vertx.redis.client package in version 3.8.3
The maxNestedArrays option is in io.vertx.redis.client.RedisOptions and not in io.vertx.redis.RedisOptions?

We are calling RedisClient.create(this.vertx, redisConfiguration) which seems to be using io.vertx.redis.RedisOptions which doesn't have the maxNestedArrays option

Please let me know if I am missing something ? Thanks.

**Comment by sg002-reactive-prog** [comment]
@pmlopes we upgraded to vertx 3.9.1
Noticed io.vertx.redis.RedisOptions is deprecated and we switched to use io.vertx.redis.client.RedisAPI to create the client

But io.vertx.redis.client.RedisOptions doesn't support SSL. This is in the Javadoc
`Does not support rediss (redis over ssl scheme) for now.`

We are using AWS elastic cache over ssl. Switching to use  io.vertx.redis.client.RedisOptionsRedisOptions completely breaks as it doesn't support SSL
we get java.io.IOException: Connection reset by peer error

Saw there is a PR for the SSL issue https://github.com/vert-x3/vertx-redis-client/pull/234
Will wait for 4.0 release to test this bug. As there is no option to set the maxNestedArrays in the old io.vertx.redis.RedisOptions class.
